core/templates/dev/head/pages/creator_dashboard/CreatorDashboard.js

```diff
@@ -452,17 +452,50 @@ oppia.controller('CreatorDashboard', [
                 oldContent: $scope.oldContent
               });
             };
+            $scope.saveSuggestion = function() {
+              $uibModalInstance.close({
+                action: ACTION_EDIT_SUGGESTION,
+                newSuggestionHtml: $scope.suggestionData.newSuggestionHtml,
+                summaryMessage: $scope.summaryMessage,
+                stateName: $scope.stateName,
+                suggestionType: $scope.suggestionType,
+                oldContent: $scope.oldContent
+              });
+            };
           }
         ]
       }).result.then(function(result) {
         var RESUBMIT_SUGGESTION_URL_TEMPLATE = (
           '/suggestionactionhandler/resubmit/<suggestion_id>');
+        var EDIT_SUGGESTION_URL_TEMPLATE = (
+          '/suggestionactionhandler/edit/<target_id>/<suggestion_id>');
         var HANDLE_SUGGESTION_URL_TEMPLATE = (
           '/suggestionactionhandler/<target_type>/<target_id>/<suggestion_id>');
 
         var url = null;
         var data = null;
-        if (result.action === 'resubmit' &&
+        if (result.action === 'edit' &&
+            result.suggestionType === 'edit_exploration_state_content') {
+          url = UrlInterpolationService.interpolateUrl(
+            EDIT_SUGGESTION_URL_TEMPLATE, {
+              target_id: $scope.activeThread.suggestion.targetId,
+              suggestion_id: $scope.activeThread.suggestion.suggestionId
+            });
+
+          data = {
+            action: result.action,
+            summary_message: result.summaryMessage,
+            change: {
+              cmd: 'edit_state_property',
+              property_name: 'content',
+              state_name: result.stateName,
+              old_value: result.oldContent,
+              new_value: {
+                html: result.newSuggestionHtml
```

Comment:
Have you tested this? (I think there should be the a content_id over here)

Reply:
Yes, I have tested it, all works fine.
But can you tell me why we use content_id 
@DubeySandeep 
thanks